### PR TITLE
Reduce `impl Into` and `impl AsRef` parameters

### PR DESCRIPTION
**Pull request by robertbastian**
It's usually better if the callers calls `x.into()` or `&x`

### CODE REVIEW

File: provider/baked/src/export.rs [hunk 1]
```diff
@@ -320,7 +317,7 @@ impl BakedExporter {
         let maybe_msrv = maybe_msrv();
 
         self.write_to_file(
-            PathBuf::from(format!("{ident}.rs.data")),
+            Path::new(&format!("{ident}.rs.data")),
```
Comment (Manishearth) [inline provider/baked/src/export.rs:320]: nice

**Review by Manishearth — APPROVED** [review]

File: utils/writeable/src/cmp.rs [hunk 2]
```diff
@@ -6,7 +6,7 @@ use core::cmp::Ordering;
 use core::fmt;
 
 pub(crate) struct WriteComparator<'a> {
-    string: &'a [u8],
+    code_points: &'a [u8],
```
Comment (sffc) [inline utils/writeable/src/cmp.rs:9]: Nit: I think I prefer this to be `string`, but did you mean `code_units` instead of `code_points`?

**Review by sffc — DISMISSED** [review]
.

File: utils/writeable/src/cmp.rs (hunk 2 above)
Comment (robertbastian) [reply utils/writeable/src/cmp.rs:9]: ah yes, `code_units`

**Review by sffc — APPROVED** [review]

**Comment by sffc** [comment]
This PR broke `bench-binsize` CI

```
error: macro expansion ends with an incomplete expression: expected expression
  --> components/icu/examples/tui.rs:82:36
   |
21 |     ($($arg:tt)*) => {};
   |                      -- in this macro arm
...
82 |             PluralCategory::One => println!("Note: You have one unread email."),
   |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected expression
   |
   = note: the macro call doesn't expand to an expression, but it can expand to a statement
help: add `;` to interpret the expansion as a statement
   |
82 |             PluralCategory::One => println!("Note: You have one unread email.");,
   |                                                                                +

error: macro expansion ends with an incomplete expression: expected expression
  --> components/icu/examples/tui.rs:83:18
   |
21 |     ($($arg:tt)*) => {};
   |                      -- in this macro arm
...
83 |             _ => println!("Note: You have {email_count} unread emails."),
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected expression
   |
   = note: the macro call doesn't expand to an expression, but it can expand to a statement
help: add `;` to interpret the expansion as a statement
   |
83 |             _ => println!("Note: You have {email_count} unread emails.");,
   |                                                                         +

error: could not compile `icu` (example "tui") due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
```

**Comment by robertbastian** [comment]
#5184

**Pull request merged**